### ISSUE TITLE
Time in events Nr. 2

### ISSUE THREAD

**Issue by pafade89** [roleoroleo/yi-hack-Allwinner-v2#817]
Hi,
in the last release @roleoroleo you change the cam to work in local time and give us the solution to chose on event page between the correct time format (#622). Unfortunately, the link still doesn't match. Wouldn't it be possible to fetch the correct time from the time zone settings on the General Settings page?

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/e5e98f18-a175-4453-bfb4-787e8ecba8d3)

I think it makes sense for the timezone and timestamps to remain consistent throughout the entire webpage.

Thank you very much for your efforts and the quick implementation last time. ;)

**Comment by roleoroleo** [comment]
Please, explain me better.

**Comment by pafade89** [comment]
My current settings for time and timezone are as follows:

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/ee302a34-881b-4ef8-81b2-af93ba543989)
![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/c14cfd65-48c1-4e9e-b32a-d1ecffec3314)

On the event page, the timestamps are now correct with GMT, so thank you for that. However, I was wondering why the timezone setting in the general settings is not used to calculate the correct time for the timestamps on the event page. I mean to say that the new setting option seems redundant in this context.

Furthermore, unfortunately, the link is still not adjusted to the now correct timestamp on the event page.

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/a588acf5-ce54-4751-a3e8-5d99d5ffe3d5)

As you can see, the link is still in local time, as far as I can tell.

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/9bfb30d8-6eed-402f-8583-fb4ed63d52a5)

Thank you in advanced

**Comment by roleoroleo** [comment]
Which model and which Yi fw do you have?

The default configuration should be correct.
Leave your timezone as above and set "Select the time used..." to autodetect.
Reboot the cam.

Now, fire an event and check:
- the datetime of the folder/file in /tmp/sd/record
- the datetime shown in Events page

Are they GMT? Or local time?
Are they different?

**Comment by pafade89** [comment]
The used model and firmware is:

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/d136f976-d672-48ee-bc64-d2ab132efd6f)

```
Leave your timezone as above and set "Select the time used..." to autodetect.
Reboot the cam.

Now, fire an event and check:
```

I have performed the steps as described by you above, and here is the result:

On the event page, it now appears that local time is being displayed:

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/ad5da89a-de54-4156-ad03-560ca9749db5)

Url in browser also shows local time:

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/3e80800b-5b78-4766-bfdc-2a4a095d243e)

FTP files path show also local time:

![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/e9bf5eee-d344-41af-a888-be6693d468e6)

To summarize, it works for local time. However, if I want to set the correct time for me (GMT), it is currently not possible for all points. The correct timestamp here would be 14:54.

**Comment by roleoroleo** [comment]
The file/folder structure in /tmp/sd/record is handled by Yi processes and I can't change this behavior.
So, the link `/?page=eventsfile&dirname=2024Y02M06D31H` can't be changed to local time.
But you should see the local time in the Events page. Is it true?

**Comment by roleoroleo** [comment]
This is not ok.
If you use the app I think cloud is enabled. Correct?
Could you try to check the time when the cloud is disabled?

**Comment by pafade89** [comment]
After turning off the cloud connection and triggering a new event, the timestamps remain unchanged, still one hour too early. Please note that we are currently in winter time, not summer time, so the offset would be even greater by an additional hour.

**Comment by roleoroleo** [comment]
Ok.
After these tests, let me rewind and restart from scratch.
I think that, with your model, you should use the GMT option in the select "Select time used by your cam...".
If I understood correctly, when you set GMT the cam works properly:
- the local time in the status is ok
- the column "Date & Time" in events page is ok
- the column "Time" in the 2nd events page is ok

The only problem is the name of the folder where the cam stores the videos.
Is it correct?
If it's correct this is the configuration for you, I can't fix the problem related to the file/folder names.
If not, please explain me what is wrong.

Another question: are you using time osd?

**Comment by pafade89** [comment]
When I use GMT option in the select "Select time used by your cam..." the following facts accurate:

 - local time on landing page is ok all the time.
 - the column "Date & Time" in events page is ok
 - the column "Time" in the 2nd events page is ok
 - the column "Directory name" in events page (time in link) is NOT ok
 - the column "File name" in the 2nd events page (time in link) is NOT ok
 - OSD time in the left corner (default and not adjustable) is ok

I can't adjust the OSD, as it always appears in the upper left corner and is therefore inevitably used. What is your question regarding OSD? Can I help you  in any way regarding that?

I have also another question :P 
What does E, M, and S stand for?
![image](https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/113902279/27947573-9084-4b79-9e6c-9d0ec94f2a49)

**Comment by pafade89** [comment]
That you are not able to change how the Yi process saves the files I understood. I only listed it again for a final summary. So, thank you for your time, and I appreciate you taking another detailed look.
Unfortunately, I can't assist with the OSD Switch since all four of my cameras always have the OSD enabled. Or can I Support you?
Otherwise, I would say, can the issue be closed?

**Comment by roleoroleo** [comment]
Don't worry about osd.
I can test it wih my cam.
But, thanks to your help I was able to test a cam with permanent osd.
And maybe, this feature is related to the offset of the file/folder structure.

You can close the issue.
Thanks.
